gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -234,6 +236,48 @@ private boolean isDirectPathEnabled(String serviceAddress) {
     return false;
   }
 
+  // DirectPath should only be used on Compute Engine. Follow this doc:
+  // https://docs.google.com/document/d/1xQXE27x9wTvwPsgiX9Hn0o8mcq5z3SKi-1jwscQsCAk/edit#.
+  private boolean isOnComputeEngine() {
+    String osName = System.getProperty("os.name");
+    Process process;
+    String result = "";
+    String tmp = null;
+    if ("Linux".equals(osName)) {
```

Comment:
It's always exactly equals to this?

Reply:
I think so, for example see https://source.corp.google.com/piper///depot/google3/java/com/google/common/flags/Flags.java;l=812?q=%22%22Linux%22.equals(osName)%22&sq=package:piper%20file:%2F%2Fdepot%2Fgoogle3%20-file:google3%2Fexperimental